source/server/hot_restarting_parent.cc

```diff
@@ -95,6 +95,11 @@ HotRestartingParent::Internal::getListenSocketsForChild(const HotRestartMessage:
   Network::Address::InstanceConstSharedPtr addr =
       Network::Utility::resolveUrl(request.pass_listen_socket().address());
   for (const auto& listener : server_->listenerManager().listeners()) {
+    if (listener.get().socket().socketType() == Network::Address::SocketType::Stream &&
```

Comment:
Pedantically, do we need the stream type check? Isn't just checking for bind to port sufficient?

Reply:
Yes, you are right, I understand it wrong, udp socket has no bind, there is no way to handle traffic, so you also need to consider, here is only need to consider whether you can bind